Решение финального задания спринта 8

IGrigoryRI opened the pull request:
Добавлены продолжительность и дата старта.
Добавлен список задач в порядке приоритета.
Добавлена проверка пересечений внутри упорядояченного. Где необходимо, циклы foreach заменены на stream API. Обновлены тесты.

Commit:
Решение финального задания спринта 8

Добавлены продолжительность и дата старта.
Добавлен список задач в порядке приоритета.
Добавлена проверка пересечений внутри упорядояченного.
Где необходимо, циклы foreach заменены на stream API.
Обновлены тесты.

Commit:
Решение финального задания спринта 8

Добавлены продолжительность и дата старта.
Добавлен список задач в порядке приоритета.
Добавлена проверка пересечений внутри упорядояченного.
Где необходимо, циклы foreach заменены на stream API.
Обновлены тесты.

Commit:
Решение финального задания спринта 8

Добавлены продолжительность и дата старта.
Добавлен список задач в порядке приоритета.
Добавлена проверка пересечений внутри упорядояченного.
Где необходимо, циклы foreach заменены на stream API.
Обновлены тесты.

Commit:
Решение финального задания спринта 8

Добавлены продолжительность и дата старта.
Добавлен список задач в порядке приоритета.
Добавлена проверка пересечений внутри упорядояченного.
Где необходимо, циклы foreach заменены на stream API.
Обновлены тесты.

kriwsha commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -108,12 +110,14 @@ private void load() {
         switch (task.getType()) {
             case TASK -> {
-                return String.format("%d,%s,%s,%s,%s,",
+                return String.format("%d,%s,%s,%s,%s,%s,%d",
```
Чтобы не усложнять менеджер подобную логику можно вынести в каждый тип задачи отдельным методом, чтобы использовать полиморфизм вместо if-ов

Commit:
Решение финального задания спринта 8

Логика работы метода taskToString из FileBackedTaskManager вынесена в каждый тип задачи

Commit:
Решение финального задания спринта 8

Логика работы метода taskToString из FileBackedTaskManager вынесена в каждый тип задачи
